instrumentation/kafka-clients/kafka-clients-0.11/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/TracingConsumerInterceptor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import java.util.Map;
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+public class TracingConsumerInterceptor<K, V> extends KafkaTracingHolder
```

Comment:
can we make this package-private, and add a method in `KafkaTracing` to create it?

```
  public <K, V> ConsumerInterceptor<K, V> newConsumerInterceptor() {
    return new TracingConsumerInterceptor<>(producerInstrumenter);
  }
```

and similar for TracingProducerInterceptor

also then we don't need `KafkaTracingHolder`

Reply:
> can we make this package-private, and add a method in KafkaTracing to create it?

No, they need to be public and have no-args constructor -- as they get created by producer/consumer via config properties.

Reply:
is there any way for the user to configure the KafkaTracing that it uses?

can you add doc to this class explaining this usage (which will also help explain to others why its public)?